src/c++/perf_analyzer/data_loader.cc

```diff
@@ -189,6 +189,9 @@ DataLoader::ParseData(
     const rapidjson::Value& steps = streams[i - offset];
     const rapidjson::Value* output_steps =
         (out_streams == nullptr) ? nullptr : &(*out_streams)[i - offset];
+
+    RETURN_IF_ERROR(ValidateParsingMode(steps));
```

Comment:
Where this is currently located, its possible the dataloader will do some to most of the processing before encountering an error, if i understand this correctly. 
Not to prematurely optimize but as we move to larger and larger models, ala LLMs, this seems like a potential location for bad user experience.
My suggestion, I would either, pull this out into its own loop and validate before processing or add a verbose print out stating "validating ..."

Reply:
All of the dataloader loading happens before any testing starts. The validation is a trivial part of that process. This function literally compares two bools.

Reply:
cool. thanks for expanding on that.